### ISSUE TITLE
Add distinct method for querysets

### ISSUE THREAD

**Issue by jamescasbon** [hmarr/mongoengine#44]
Please wrap pymongo's distinct.  Something like: 

```
def distinct(self, foo):
    return self._cursor.distinct(foo)
```

but you need to transform foo into a pymongo query string.

**Comment by iapain** [comment]
It is not compatible with pymongo 1.1, hence we need to look out at dependencies first.

**Comment by hmarr** [comment]
Added QuerySet.distinct. Closed by 9f98025b8c13d9e845f5de68e29e825f5517b245.
